blocks/soapy/include/gnuradio-4.0/soapy/SoapyRaiiWrapper.hpp

```diff
@@ -0,0 +1,805 @@
+        for (size_t i = 0; i < _cArgs.size; ++i) {
+            delete[] _cArgs.keys[i];
+            delete[] _cArgs.vals[i];
+        }
+        delete[] _cArgs.keys;
+        delete[] _cArgs.vals;
+    }
+
+    [[nodiscard]] SoapySDRKwargs* get() noexcept { return &_cArgs; }
+    operator const SoapySDRKwargs*() const noexcept { return &_cArgs; }
+    operator SoapySDRKwargs*() noexcept { return &_cArgs; }
+};
+
+[[nodiscard]] inline SoapySDR::KwargsList convertToCpp(SoapySDRKwargs* results, std::size_t length) {
+    SoapySDR::KwargsList kwargsList;
+    kwargsList.reserve(length);
+    for (size_t i = 0; i < length; ++i) {
+        SoapySDR::Kwargs kwargs;
+        for (size_t j = 0; j < results[i].size; ++j) {
+            kwargs[results[i].keys[j]] = results[i].vals[j];
+        }
+        kwargsList.push_back(kwargs);
+        SoapySDRKwargs_clear(&(results[i]));
+    }
+    SoapySDR_free(results);
+    return kwargsList;
+}
+
+[[nodiscard]] inline std::vector<std::string> convertToCpp(char** strings, std::size_t length) {
+    std::vector<std::string> stringList;
+    stringList.reserve(length);
+    for (size_t i = 0; i < length; ++i) {
+        stringList.emplace_back(std::string(strings[i]));
+    }
+    SoapySDRStrings_clear(&strings, length);
+    return stringList;
+}
+
+[[nodiscard]] inline std::vector<ArgInfo> convertToCpp(SoapySDRArgInfo* infos, std::size_t size) {
+    std::vector<ArgInfo> argInfoList;
+    argInfoList.resize(size); // Preallocate space
+
+    for (std::size_t i = 0; i < size; ++i) {
+        SoapySDRArgInfo& cInfo   = infos[i];
+        ArgInfo&         argInfo = argInfoList[i];
+
+        argInfo.key         = cInfo.key ? std::string(cInfo.key) : "";
+        argInfo.value       = cInfo.value ? std::string(cInfo.value) : "";
+        argInfo.name        = cInfo.name ? std::string(cInfo.name) : "";
+        argInfo.description = cInfo.description ? std::string(cInfo.description) : "";
+        argInfo.units       = cInfo.units ? std::string(cInfo.units) : "";
+        argInfo.type        = static_cast<ArgInfo::Type>(cInfo.type);
+        argInfo.range       = {cInfo.range.minimum, cInfo.range.maximum, cInfo.range.step};
+
+        if (cInfo.options != nullptr) {
+            argInfo.options.resize(cInfo.numOptions);
+            for (size_t j = 0; j < cInfo.numOptions; ++j) {
+                assert(cInfo.options[j] != nullptr && "Option string should not be null");
+                argInfo.options[j] = std::string(cInfo.options[j]);
+            }
+        }
+
+        if (cInfo.optionNames != nullptr) {
+            argInfo.optionNames.resize(cInfo.numOptions);
+            for (size_t j = 0; j < cInfo.numOptions; ++j) {
+                assert(cInfo.optionNames[j] != nullptr && "Option name string should not be null");
+                argInfo.optionNames[j] = std::string(cInfo.optionNames[j]);
+            }
+        }
+
+        SoapySDRArgInfo_clear(&cInfo);
+    }
+    SoapySDR_free(infos);
```

Comment:
```suggestion
    }
    SoapySDRArgInfoList_clear(infos, size);
```
There is a function to clear an array of `SoapySDRArgInfo`. Internally it does the same. However, address sanitizer still indicates some memory leaks.

Reply:
done